wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -361,53 +355,95 @@ instance Arbitrary PasswordUpdate where
-data EstimatedFees = EstimatedFees
-  { -- | The estimated fees, as coins.
+-- | 'EstimatedFees' represents the fees which would be generated
+-- for a 'Payment' in case the latter would actually be performed.
+data EstimatedFees = EstimatedFees {
     feeEstimatedAmount :: !Coins
+    -- ^ The estimated fees, as coins.
   } deriving (Show, Eq, Generic)
 
 deriveJSON Serokell.defaultOptions ''EstimatedFees
 
 instance Arbitrary EstimatedFees where
   arbitrary = EstimatedFees <$> fmap getPositive arbitrary
 
--- | Stub type for a `Payment`.
+-- | Maps an 'Address' to some 'Coin's, and it's
+-- typically used to specify where to send money during a 'Payment'.
+data PaymentDistribution = PaymentDistribution {
+      pdAddress :: Core.Address
+    , pdAmount  :: Core.Coin
+    } deriving (Show, Eq)
+
+deriveJSON Serokell.defaultOptions ''PaymentDistribution
+
+instance Arbitrary PaymentDistribution where
+  arbitrary = PaymentDistribution <$> arbitrary
+                                  <*> arbitrary
+
+-- | A policy to be passed to each new payment request to
+-- determine how a 'Transaction' is assembled.
+data TransactionGroupingPolicy =
+      OptimiseForSizePolicy
+    -- ^ Tries to minimise the size of the created transaction
+    -- by choosing only the biggest value available up until
+    -- the stake sum is greater or equal the payment amount.
+    | OptimiseForSecurityPolicy
+    -- ^ Tries to minimise the number of addresses left with
+    -- unspent funds after the transaction has been created.
+    deriving (Show, Ord, Eq, Enum, Bounded)
+
+instance Arbitrary TransactionGroupingPolicy where
+  arbitrary = elements [minBound .. maxBound]
+
+-- Drops the @Policy@ suffix.
+deriveJSON defaultOptions { constructorTagModifier = reverse . drop 6 . reverse } ''TransactionGroupingPolicy
+
+-- | A 'Payment' from one source account to one or more 'PaymentDistribution'(s).
 data Payment = Payment
-  { -- | The source Account.
-    pmtSourceAccount      :: !Account
-    -- | The destination Address.
-  , pmtDestinationAddress :: !Address
-    -- | The amount for this payment.
-  , pmtAmount             :: !Coins
+  { pmtSourceWallet   :: !WalletId
+    -- ^ The source Wallet.
+  , pmtSourceAccount  :: !AccountId
+    -- ^ The source Account.
+  , pmtDestinations   :: !(NonEmpty PaymentDistribution)
+    -- ^ The destinations for this payment.
+  , pmtGroupingPolicy :: !(Maybe TransactionGroupingPolicy)
+    -- ^ Which strategy use in grouping the input transactions.
   } deriving (Show, Eq, Generic)
 
 deriveJSON Serokell.defaultOptions ''Payment
 
 instance Arbitrary Payment where
   arbitrary = Payment <$> arbitrary
                       <*> arbitrary
-                      <*> fmap getPositive arbitrary
+                      <*> arbitrary
+                      <*> arbitrary
 
 type TxId = Text
 
--- | A Wallet Transaction.
+-- | A 'Wallet''s 'Transaction'.
+-- TODO(adinapoli) This still doesn't include the full payload frontend might
```

Comment:
Indeed. Probably we should think about adding such data as `isOutgoing` flag right now?

Reply:
Not sure about this last one as I would like to see maybe if we can coalesce `isOutgoing` and `isLocal` into a single datatype. Do you think it's a good idea or we should just stick with the 2 separated booleans? I was thinking something along the line of (I suck at names, feel free to adjust/propose):

```
data TransactionType =
    LocalTransaction
  | OutgoingTransaction
```

and then have something like (in the final json):

```
{
  "type": "local"
}
```

Terrible idea?

Reply:
I like this idea, though then there should be three types of transaction:
```
data TransactionType =
    LocalTransaction
  | IncomingTransaction
  | OutgoingTransaction
```
I believe though that you had that in mind, just accidentally forgot one line :).

Reply:
I like it, let's do it 😉